File: Code Chef/JavaScript/700-799/739 Building Race BUILDINGRACE.js
let input = '';
process.stdin.on('data', (data) => input += data);

const strToInt = (strArray) => {
    strArray.shift();
    const intArray = strArray.map((strItem) => strItem.trim().split(' ').map(Number));
    
    return intArray;
};

const whoWillWin = (floorOfChef, floorOfChefina, speedOfChef, speedOfChefina) => {
    
    const timeOfChef = floorOfChef / speedOfChef;
    const timeOfChefina = floorOfChefina / speedOfChefina;
    
    const winner = (timeOfChef === timeOfChefina)
        ? 'Both'
        : (timeOfChef < timeOfChefina)
            ? 'Chef' : 'Chefina';
    
    console.log(winner);
};

process.stdin.on('end', () => {
    const testcases = strToInt(input.trim().split('\n'));
    
    testcases.forEach((testcase) => {
        const [floorOfChef, floorOfChefina, speedOfChef, speedOfChefina] = testcase;
        whoWillWin(floorOfChef, floorOfChefina, speedOfChef, speedOfChefina);
    });
});

File: Code Chef/C Programming/700-799/739 Building Race BUILDINGRACE.c
#include <stdio.h>
int main(void) {
	int t;
	float a,b,x,y;
	scanf("%d",&t);
	for(int i=0;i<t;i++)
	{
	    scanf("%f %f %f %f",&a,&b,&x,&y);
	    float c=a/x,d=b/y;
	    if(c==d)
	    {
	        printf("Both\n");
	    }
	    else if(c>d)
	    {
	        printf("Chefina\n");
	    }
	    else if(c<d)
	    {
	        printf("Chef\n");
	    }
	}
	return 0;
}